ethereumj-core/src/main/java/org/ethereum/config/blockchain/ErpConfig.java

```diff
@@ -0,0 +1,167 @@
+            throw new RuntimeException(e);
+        }
+    }
+
+    void initErpConfig() throws IOException {
+        // load the config block numbers
+        final Collection<ErpMetadata> allErps = erpLoader.loadErpMetadata();
+        this.erpDataByTargetBlock = allErps
+            .stream()
+            .collect(toMap(ErpMetadata::getTargetBlock, Function.identity()));
+
+        logger.info("Found %d ERPs", allErps.size());
+
+        // add the header validators for each known ERP
+        final List<Pair<Long, BlockHeaderValidator>> headerValidators = headerValidators();
+        allErps.forEach(erpMetadata -> {
+            BlockHeaderRule rule = new ExtraDataPresenceRule(erpMetadata.getErpMarker(), true);
+            headerValidators.add(Pair.of(erpMetadata.getTargetBlock(), new BlockHeaderValidator(rule)));
+        });
+    }
+
+    /**
+     * Miners should include marker for initial 10 blocks. Either "dao-hard-fork" or ""
+     */
+    @Override
+    public byte[] getExtraData(byte[] minerExtraData, long blockNumber) {
+        // TODO: is EXTRA_DATA_AFFECTS_BLOCKS_NUMBER needed?
+        final ErpMetadata erpMetadata = erpDataByTargetBlock.get(blockNumber);
+        return erpMetadata != null
+            ? erpMetadata.getErpMarker()
+            : minerExtraData;
+    }
+
+    @Override
+    public void hardForkTransfers(Block block, Repository repo) {
+        final ErpMetadata erpMetadata = erpDataByTargetBlock.get(block.getNumber());
+        if (erpMetadata != null) {
+            logger.info("Found ERP {} for block {}", erpMetadata.getId(), erpMetadata.getTargetBlock());
+            doHardForkTransfers(erpMetadata, repo);
+        }
+    }
+
+    void doHardForkTransfers(ErpMetadata erpMetadata, Repository repo) {
+        final StateChangeObject sco;
+        try {
+            sco = erpLoader.loadStateChangeObject(erpMetadata);
+        } catch (IOException e) {
+            logger.error("Failed to load state change object for {}", erpMetadata.getId(), e);
+            throw new RuntimeException("Failed to load state change object for " + erpMetadata.getId(), e);
+        }
+
+        // TODO: Is this the right way to apply changes in  batch?
+        final Repository track = repo.startTracking();
+        try {
+            erpExecutor.applyStateChanges(sco, track);
+            track.commit();
+            logger.info("Successfully applied ERP '{}' to block {}", erpMetadata.getId(), erpMetadata.getTargetBlock());
+        }
+        catch (ErpExecutor.ErpExecutionException e) {
+            track.rollback();
+            logger.error("Failed to apply ERP '{}' to block {}", erpMetadata.getId(), erpMetadata.getTargetBlock(), e);
+        }
+        catch (Exception e) {
+            track.rollback();
+            logger.error("Failed to apply ERP '{}' to block {}", erpMetadata.getId(), erpMetadata.getTargetBlock(), e);
+            throw e;
+        }
+        finally {
+            track.close();
```

Comment:
I was thinking it would actually be better to leave it.  The startTracking method on the Repository interface returns another Repository; it makes no promise about the implementation.  Whether the close method is a no-op or not is an implementation detail of the returned class, right?

Reply:
ok, let's keep it there